src/common/ios/iOSPlatform.ts

```diff
@@ -1,140 +1,161 @@
-                    });
-            });
-    }
-
-    public prewarmBundleCache(): Q.Promise<void> {
-        return this.remoteExtension.prewarmBundleCache(this.platformName);
-    }
-
-    private generateSuccessPatterns(): Q.Promise<string[]> {
-        return this.getBundleId().then(bundleId =>
-            IOSPlatform.RUN_IOS_SUCCESS_PATTERNS.concat([`Launching ${bundleId}\n${bundleId}: `]));
-    }
-
-    private getBundleId(): Q.Promise<string> {
-        return this.plistBuddy.getBundleId(this.iosProjectPath);
-    }
-}
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import * as path from "path";
+
+import {Log} from "../../common/log/log";
+import {ChildProcess} from "../../common/node/childProcess";
+import {CommandExecutor} from "../../common/commandExecutor";
+import {GeneralMobilePlatform} from "../../common/generalMobilePlatform";
+import {IRunOptions} from "../../common/launchArgs";
+import {PlistBuddy} from "../../common/ios/plistBuddy";
+import {IOSDebugModeManager} from "../../common/ios/iOSDebugModeManager";
+import {OutputVerifier, PatternToFailure} from "../../common/outputVerifier";
+import {RemoteExtension} from "../../common/remoteExtension";
+
+/* tslint:disable:no-var-requires */
+const ErrorStrings = require("../../../errorStrings/errorStrings.json");
+/* tslint:enable:no-var-requires */
+
+export class IOSPlatform extends GeneralMobilePlatform {
+    public static DEFAULT_IOS_PROJECT_RELATIVE_PATH = "ios";
+    public static DEFAULT_IOS_SIMULATOR_TARGET = "iPhone 5";
+
+    private static deviceString = "device";
+    private static simulatorString = "simulator";
+
+    private plistBuddy = new PlistBuddy();
+    private target: string = "";
+    private targetType: string = "simulator";
+    private iosProjectPath: string;
+
+    // We should add the common iOS build/run erros we find to this list
+    private static RUN_IOS_FAILURE_PATTERNS: PatternToFailure[] = [{
+        pattern: "No devices are booted",
+        message: ErrorStrings.IOSSimulatorNotLaunchable,
+    }, {
+        pattern: "FBSOpenApplicationErrorDomain",
+        message: ErrorStrings.IOSSimulatorNotLaunchable,
+    }, {
+        pattern: "ios-deploy",
+        message: ErrorStrings.IOSDeployNotFound,
+    }];
+
+    private static RUN_IOS_SUCCESS_PATTERNS = ["BUILD SUCCEEDED"];
+
+    // We set remoteExtension = null so that if there is an instance of iOSPlatform that wants to have it's custom remoteExtension it can. This is specifically useful for tests.
+    constructor(runOptions: IRunOptions, { remoteExtension = undefined }: {remoteExtension?: RemoteExtension} = {}) {
+        super(runOptions, { remoteExtension: remoteExtension });
+
+        if (this.runOptions.targetType && (this.runOptions.targetType !== IOSPlatform.simulatorString && this.runOptions.targetType !== IOSPlatform.deviceString)) {
```

Comment:
For the sake of readability consider splitting this into a multiline expression :)